packages/create-wagmi/templates/vite-react/src/main.tsx

```diff
@@ -0,0 +1,56 @@
+import * as React from 'react'
+import * as ReactDOM from 'react-dom/client'
+
+import {
+  WagmiConfig,
+  configureChains,
+  createClient,
+  defaultChains,
+} from 'wagmi'
+import { CoinbaseWalletConnector } from 'wagmi/connectors/coinbaseWallet'
+import { InjectedConnector } from 'wagmi/connectors/injected'
+import { MetaMaskConnector } from 'wagmi/connectors/metaMask'
+import { WalletConnectConnector } from 'wagmi/connectors/walletConnect'
+import { alchemyProvider } from 'wagmi/providers/alchemy'
+
+import { App } from './App'
+
+const { chains, provider, webSocketProvider } = configureChains(defaultChains, [
+  alchemyProvider({ apiKey: import.meta.env.VITE_ALCHEMY_API_KEY as string }),
+])
+
+const client = createClient({
+  autoConnect: true,
+  connectors: [
+    new MetaMaskConnector({ chains }),
+    new CoinbaseWalletConnector({
+      chains,
+      options: {
+        appName: 'wagmi',
+      },
+    }),
+    new WalletConnectConnector({
+      chains,
+      options: {
+        qrcode: true,
+      },
+    }),
+    new InjectedConnector({
+      chains,
+      options: {
+        name: 'Injected',
+        shimDisconnect: true,
+      },
+    }),
+  ],
+  provider,
+  webSocketProvider,
+})
```

Comment:
wdyt about breaking out all this wagmi code into a `src/wagmi.ts` file for all the templates? cleans up the entrypoint (e.g. `main.tsx`, `_app.tsx`) and puts all of it in a dedicated area.

Also, will allow us to more easily control customization during set up in the future (select chains, RPC provider, connectors, etc).